### PR TITLE
fix: add metadata arg to BlockPayload::transaction_commitments

### LINKED ISSUE #2398: [API] - BlockPayload::transaction_commitments needs a metadata arg
### What is this task and why do we need to work on it?

We forgot to add an arg `metadata: &Self::Metadata` to the `transaction_commitments` method of the `BlockPayload` trait.

This new arg is needed downstream in `espresso-sequencer` to enable use of the updated `QueryPayload` trait from the `hotshot-query-service` repo. Example:

https://github.com/EspressoSystems/espresso-sequencer/blob/e177c1e2434193f3905e927a163afbb2aaf8f72c/sequencer/src/block.rs#L345-L347

The new `enumerate` call requires a `metadata` arg.

### What work will need to be done to complete this task?

Add the arg to the trait method, then fix everything that breaks in this repo and downstream.

Also, there are a few downstream build warnings due to `hotshot-testing` feature flag that could be fixed here too.

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

Nothing else.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by ggutoski**
Closes #2398 

### This PR: 

- Add `metadata` arg to `BlockPayload::transaction_commitments` and fix everything that breaks.
- Fix a few downstream build warnings due to `hotshot-testing` feature flag.

### This PR does not: 

...do anything else

### Key places to review: 

- Core API change in `block_contents.rs`
- All changes in `task-impls/` and `testing/` is an immediate consequence of this API change.
- `lib.rs` and `handle.rs`: fix downstream build warnings and tidy `use` statements.

### CODE REVIEW

**Comment by ggutoski** [comment]
Forgot to mention: After this PR is merged, in order to use this fix downstream in espresso-sequencer we need:
- a new version tag 0.5.whatever
- update hotshot dependency of hotshot-query-service
- update hotshot, hotshot-query-service dependencies of espresso-sequencer

**Comment by ggutoski** [comment]
cc @akonring fixed the final few build errors with some help from @nomaxg . This PR is the next step.

**Comment by ggutoski** [comment]
libp2p test failure in CI. Everything passes locally. 🤷

File: crates/hotshot/src/lib.rs [hunk 1]
```diff
@@ -256,6 +261,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
 
         let inner: Arc<SystemContextInner<TYPES, I>> = Arc::new(SystemContextInner {
             id: nonce,
+            #[cfg(feature = "hotshot-testing")]
```
Comment (elliedavidson) [inline crates/hotshot/src/lib.rs:264]: @shenkeyao Can we make an issue to delete this field?  It looks like we don't use it anymore.

**Review by elliedavidson — APPROVED** [review]

File: crates/hotshot/src/lib.rs (hunk 1 above)
Comment (ggutoski) [reply crates/hotshot/src/lib.rs:264]: While you're at it... there's another field in this struct called `_metrics` that also appears not to be in use. Maybe we could delete it, too. 🙏

Comment (shenkeyao) [reply crates/hotshot/src/lib.rs:264]: Sure! Issue created: https://github.com/EspressoSystems/HotShot/issues/2406.

**Pull request merged**